nautobot/ipam/tests/test_filters.py

```diff
@@ -277,304 +222,180 @@ def test_prefix(self):
+        )
+        vrfs = (
+            VRF.objects.create(name="VRF 1", rd="65000:100"),
+            VRF.objects.create(name="VRF 2", rd="65000:200"),
+            VRF.objects.create(name="VRF 3", rd="65000:300"),
+        )
+        vrfs[0].import_targets.add(route_targets[0], route_targets[1], route_targets[2])
+        vrfs[1].export_targets.add(route_targets[1])
+        vrfs[2].export_targets.add(route_targets[2])
+        Prefix.objects.filter(pk__in=[test_prefixes[0].pk, test_prefixes[1].pk]).update(vrf=vrfs[0])
+        Prefix.objects.filter(pk__in=[test_prefixes[2].pk, test_prefixes[3].pk]).update(vrf=vrfs[1])
+        Prefix.objects.filter(pk__in=[test_prefixes[4].pk, test_prefixes[5].pk]).update(vrf=vrfs[2])
+        self.assertEqual(self.filterset({"present_in_vrf_id": vrfs[0].pk}, self.queryset).qs.count(), 6)
+        self.assertEqual(self.filterset({"present_in_vrf_id": vrfs[1].pk}, self.queryset).qs.count(), 2)
+        self.assertEqual(self.filterset({"present_in_vrf": vrfs[0].rd}, self.queryset).qs.count(), 6)
+        self.assertEqual(self.filterset({"present_in_vrf": vrfs[1].rd}, self.queryset).qs.count(), 2)
 
     def test_region(self):
-        regions = Region.objects.all()[:2]
+        regions = {Prefix.objects.filter(site__region__isnull=False).values_list("site__region", flat=True)}
+        self.assertSequenceEqual([], [regions])
+        if len(regions) < 2:
+            unique_site = Site.objects.filter(region__isnull=False).exclude(region__in=regions).first()
+            PrefixFactory(site=unique_site)
+            regions.add(unique_site.region)
         params = {"region_id": [regions[0].pk, regions[1].pk]}
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
         params = {"region": [regions[0].slug, regions[1].slug]}
         self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
 
-    def test_site(self):
-        sites = Site.objects.all()[:2]
-        params = {"site_id": [sites[0].pk, sites[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
-        params = {"site": [sites[0].slug, sites[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
-
-    def test_vlan(self):
-        vlans = VLAN.objects.all()[:2]
-        params = {"vlan_id": [vlans[0].pk, vlans[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
-        # TODO: Test for multiple values
-        params = {"vlan_vid": vlans[0].vid}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 2)
 
-    def test_role(self):
-        roles = Role.objects.all()[:2]
-        params = {"role_id": [roles[0].pk, roles[1].pk]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
-        params = {"role": [roles[0].slug, roles[1].slug]}
-        self.assertEqual(self.filterset(params, self.queryset).qs.count(), 4)
+# def test_site(self):
```

Comment:
Are these deprecated or will these be restored?

Reply:
This file wasn't ready for review yet